### ISSUE TITLE
Off-by-one error in search result highlighting

### ISSUE THREAD

**Issue by lennartkoopmann** [Graylog2/graylog2-server#2505]
### Problem description

When searching with search result highlighting on, the actual UI highlighting seems to be off by one index:

![screen shot 2016-07-19 at 5 18 58 pm](https://cloud.githubusercontent.com/assets/35022/16968531/104f66fc-4dd5-11e6-81b2-bf2f08740930.png)

This happens with both full text searches and direct field matches.

### Environment

* Graylog Version: Graylog 2.1.0-alpha.3-SNAPSHOT+53aade9
* Elasticsearch Version: 2.3.1
* MongoDB Version: 3.0.0
* Operating System: OSX
* Browser version: n/a

**Comment by joschi** [comment]
@lennart I couldn't reproduce this using Graylog 2.1.0-alpha.3-SNAPSHOT+97521bf2bd8170101173aa6c39a1cf025e6ab1e8.

<img width="1435" alt="issue-2505-chrome" src="https://cloud.githubusercontent.com/assets/43951/17077483/1dce99ba-50d2-11e6-955f-7d0ec76772b9.png">
<img width="1352" alt="issue-2505-safari" src="https://cloud.githubusercontent.com/assets/43951/17077486/28694bc2-50d2-11e6-8146-bd4578cf47c3.png">


I'm closing this ticket as I can't reproduce the described problem. Please re-open the ticket if you find this happening again.

**Comment by shanexu** [comment]
We have this problem too. And I can reproduce it. The message should has leading spaces. 
In the org.graylog2.plugin.Message class, the method addField, always trim string value, then the highlightRanges are wrong. The issues should be reopened.

```java
    private void addField(final String key, final Object value, final boolean isRequiredField) {
        final String trimmedKey = key.trim();

        // Don't accept protected keys. (some are allowed though lol)
        if (RESERVED_FIELDS.contains(trimmedKey) && !RESERVED_SETTABLE_FIELDS.contains(trimmedKey) || !validKey(trimmedKey)) {
            if (LOG.isDebugEnabled()) {
                LOG.debug("Ignoring invalid or reserved key {} for message {}", trimmedKey, getId());
            }
            return;
        }

        if (FIELD_TIMESTAMP.equals(trimmedKey) && value != null && value instanceof Date) {
            fields.put(FIELD_TIMESTAMP, new DateTime(value));
        } else if (value instanceof String) {
            final String str = ((String) value).trim();

            if (isRequiredField || !str.isEmpty()) {
                fields.put(trimmedKey, str);
            }
        } else if (value != null) {
            fields.put(trimmedKey, value);
        }
    }
```

**Comment by joschi** [comment]
@shanexu Thanks for your feedback!

This issue has been addressed in #2510 and will lead to the expected behavior in Graylog 2.1.0. There will probably be no fix for Graylog 2.0.x because it was a breaking change.
